### PR TITLE
vault: Common HTTP response handling - prepared for JSON

### PR DESCRIPTION
**Pull request by cfconrad**
We always expect JSON as response to our REST API calls.
So we can make a common error handling for httpGet() and httpPost(),
which return json object or raise an exception on error.

### CODE REVIEW

File: tests/test_vault.py [hunk 1]
```diff
@@ -21,13 +21,21 @@
...
+    def __init__(self, json_response=None, status_code=200, content_type='application/json', url='not_given', text=""):
         self.status_code = status_code
+        if json_response is not None:
+            text = json.dumps(json_response)
+        self.content = text
+        self.text = text
+        self.headers = {'content-type': content_type}
+        self.url = url
 
     def json(self):
```
Comment (asmorodskyi) [inline tests/test_vault.py]: where we using this method ?

**Comment by asmorodskyi** [comment]
generally looks good to me , just have concerns about `MockResponse.json` ( see my comment in code )

File: tests/test_vault.py (hunk 1 above)
Comment (cfconrad) [reply tests/test_vault.py]: We are using it here: https://github.com/cfconrad/pcw/pull/68/files#diff-7d35fc4e9f2b0c45d5681b1c0bbc34a8R77

Comment (asmorodskyi) [reply tests/test_vault.py]: ah ok I just didn't realize that we overloading this method so I need to check prod code and not test code

**Pull request merged**